modules/acquia_telemetry/src/Telemetry.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Drupal\acquia_telemetry;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Component\Utility\Crypt;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleExtensionList;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Telemetry service.
+ */
+class Telemetry {
+
+  /**
+   * Amplitude API URL.
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  const AMPLITUDE_API_URL = 'https://api.amplitude.com/httpapi';
+
+  /**
+   * Amplitude API key.
+   *
+   * This is not intended to be private. It is typically included in client
+   * side code. Fetching data requires an additional API secret.
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  const AMPLITUDE_API_KEY = 'f32aacddde42ad34f5a3078a621f37a9';
+
+  /**
+   * The extension.list.module service.
+   *
+   * @var \Drupal\Core\Extension\ModuleExtensionList
+   */
+  protected $extensionListModule;
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * The config.factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config_factory;
+
+  /**
+   * Constructs a telemetry object.
+   *
+   * @param \Drupal\Core\Extension\ModuleExtensionList $extension_list_module
+   *   The extension.list.module service.
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config.factory service.
+   */
+  public function __construct(ModuleExtensionList $extension_list_module, ClientInterface $http_client, ConfigFactoryInterface $config_factory) {
+    $this->extensionListModule = $extension_list_module;
```

Comment:
Nit: I think $extensionListModule is an awkward name for the property. How about just $moduleList?